torch/_dynamo/variables/builder.py

```diff
@@ -440,6 +440,31 @@ def index_source(key):
                 guards=make_guards(GuardBuilder.FUNCTION_MATCH),
             )
         # NB: These can't be put in type_dispatch, they have to run later
+        elif value in {
+            torch.distributed.distributed_c10d.all_gather_into_tensor,
+            torch.distributed.distributed_c10d.reduce_scatter_tensor,
+        }:
+            from torch.distributed._functional_collectives import (
+                all_gather_tensor_inplace,
+                reduce_scatter_tensor_inplace,
+            )
+            from torch.distributed.distributed_c10d import (
+                all_gather_into_tensor as legacy_allgather,
+                reduce_scatter_tensor as legacy_reducescatter,
+            )
+
+            # TODO assert that async_op is false, and, possibly some other limitations
+            # current remapping is very brittle, in regards to handling args at call time
+            # and seriously.. can't import properly, and also getting line-too-long linter
+            traceable_collective_remaps = {
+                legacy_allgather: all_gather_tensor_inplace,
+                legacy_reducescatter: reduce_scatter_tensor_inplace,
+            }
```

Comment:
If you build this remaps outside the if/elif chain here, you can remove most of the function here

```
traceable_collective_remaps = {
torch.distributed.distributed_c10d.all_gather_tensor_inplace : torch.distributed._functional_collectives.all_gather_tensor_inplace, 
torch.distributed.distributed_c10d.reduce_scatter_tensor_inplace : torch.distributed._functional_collectives.reduce_scatter_tensor_inplace, 
}
```

```
elif value in traceable_collective_remaps:
    return UserFunctionVariable(
                traceable_collective_remaps[value],
                source=self.source,
                guards=make_guards(GuardBuilder.FUNCTION_MATCH),
            )
```

Reply:
yea. i intend to put this dict in _functional_collectives.py, and then use it for the elif check as well as the mapping.

but i ran into circular import hell.

i didn't figure that part out yet but ill get there